### PR TITLE
Prereq

### PR DESCRIPTION
**Pull request by cubetastic33**
Parse pre-reqs like "CS 1337 with a C" using regex

image

### CODE REVIEW

**Comment by jasonappah** [comment]
Looks good to me, thanks for the PR!

**Review by jasonappah — DISMISSED** [review]

File: src/server/trpc/router/validator.ts [hunk 1]
```diff
@@ -77,6 +77,9 @@ export const validatorRouter = router({
       const coReqHash = new Map<string, Array<[Array<string>, number]>>();
       const preReqHash = new Map<string, Array<[Array<string>, number]>>();
       const coOrPreReqHash = new Map<string, Array<[Array<string>, number]>>();
+      // Regex to parse course from description of improperly parsed course
+      const re = /\b[A-Z]{2} \d{4}\b/;
```
Comment (ZzRanger) [inline src/server/trpc/router/validator.ts:81]: Some course codes can be longer than 2 characters (i.e. HIST 3301) so we should ideally account for that

**Review by ZzRanger — CHANGES_REQUESTED** [review]
After that change we'll look to merge this. Once we have JIRA set up and everything we'll make a second issue to start writing tests for prereq validation as well as monitor the progress of the new coursebook scaper.

File: src/server/trpc/router/validator.ts (hunk 1 above)
Comment (cubetastic33) [reply src/server/trpc/router/validator.ts]: Oh righttt I completely forgot, fixed it now so it accepts 2-4 chars (so things like ACN will parse too)

**Review by akevinge — APPROVED** [review]

**Review by jasonappah — APPROVED** [review]

**Review by ZzRanger — APPROVED** [review]

**Pull request merged**